packages/react-dom/src/events/DOMEventResponderSystem.js

```diff
@@ -990,3 +998,29 @@ export function generateListeningKey(
   const passiveKey = passive ? '_passive' : '_active';
   return `${topLevelType}${passiveKey}`;
 }
+
+let lastDiscreteEventTimeStamp = 0;
+let useTimeStampHeuristic = true;
+
+if (canUseDOM) {
+  const firefoxUA = navigator.userAgent.match(/Firefox\/(.*)$/);
```

Comment:
What will the behavior be in Firefox then? Will it be broken, or merely less optimal?

Also perhaps there's a way to detect this that doesn't rely on UA sniffing. Like by dispatching a click event on a dummy node and checking the timestamps of the events.

Reply:
Actually, I just managed to test on FF 51 again, we don't need this anymore. The timestamps might be different resolutions, as pointed out with Vue, but we don't care for the relevance in terms of difference (> or <), we only care for the equality (===).